### PR TITLE
✨ add Dropbox Business provider

### PR DESCRIPTION
**Pull request by chris-rock**
Adds the **dropbox** provider for Dropbox Business teams, using the vendor `dropbox-sdk-go-unofficial/v6` SDK with a team access token (`--token` or `DROPBOX_TEAM_TOKEN`).

**Resources:** `dropbox`, `dropbox.team`, `dropbox.member`, `dropbox.group`, `dropbox.device`, `dropbox.linkedApp`.

Implements **ADR-039** (included). Registered in `defaults.go`, `README`, `Makefile`, and the `go.work` list. Builds green.

Follow-up: extend the team resource to surface 2FA/SSO settings (`team/features/get_values` + member profiles) so the cnspec policy can check them.

### CODE REVIEW

File: providers/dropbox/connection/connection.go [hunk 1]
```diff
@@ -0,0 +1,198 @@
...
+	if c.teamInfo != nil {
+		return c.teamInfo, nil
+	}
+	info, err := c.client.GetInfo()
+	if err != nil {
+		return nil, err
+	}
+	c.teamInfo = info
+	return info, nil
+}
```
[the remark this answers is not in the document]
Comment (chris-rock) [reply providers/dropbox/connection/connection.go:138]: Fixed in `0b6437d`. `TeamInfo()` now caches through a `sync.Once` (`teamInfoOnce` + `teamInfoErr`), matching the `DevicesByMember`/`LinkedAppsByMember` pattern in the same file. Behavior is preserved: if `Verify()` already populated `teamInfo`, the `Do` block returns early and reuses it.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: providers/dropbox/resources/dropbox.lr [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+  numLicensedUsers int
+  // Number of accounts invited to or already active on the team
+  numProvisionedUsers int
+  // Number of licenses currently used by the team
+  numUsedLicenses int
+  // Whether shared folders may include members from outside the team
+  externalSharingAllowed bool
+  // Whether shared links default to a publicly accessible audience
+  publicSharingAllowed bool
+  // Upload API rate limit in calls per month, 0 when unlimited or not configured
```
[the remark this answers is not in the document]
Comment (chris-rock) [reply providers/dropbox/resources/dropbox.lr:50]: Fixed in `384d6da`. Aligned the doc to the "permitted" semantics the code implements. The field comment now reads "Whether team members are permitted to create shared links with a publicly accessible audience", and the resource-level doc was updated to match. Code is unchanged: `default_team_only` stays in the `true` set because members can still create public links under that policy, only `team_only` and `default_no_one` bar them.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: providers/dropbox/resources/member.go [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+
+	secondaryEmails := make([]any, 0, len(profile.SecondaryEmails))
+	for _, se := range profile.SecondaryEmails {
+		if se != nil {
+			secondaryEmails = append(secondaryEmails, se.Email)
+		}
+	}
+
+	res, err := CreateResource(runtime, "dropbox.member", map[string]*llx.RawData{
+		"__id":            llx.StringData(profile.TeamMemberId),
```
[the remark this answers is not in the document]
Comment (chris-rock) [reply providers/dropbox/resources/member.go]: Fixed in `384d6da`. The `__id` args now carry the same prefixed value that `id()` returns, so the two agree: `dropbox.member/<id>`, `dropbox.device/<memberId>/<sessionId>`, and `dropbox.linkedApp/<memberId>/<appId>`. This closes the latent cache-miss gap if any future path calls `NewResource` without `__id`.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: providers/dropbox/resources/dropbox.go [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+func (r *mqlDropbox) conn() *connection.DropboxConnection {
+	return r.MqlRuntime.Connection.(*connection.DropboxConnection)
+}
+
+// initDropboxTeam reads the team-wide identity and sharing configuration
+// singleton. There is exactly one team per connection, so it resolves entirely
+// from the connection and uses the team ID as its cache key.
+func initDropboxTeam(runtime *plugin.Runtime, args map[string]*llx.RawData) (map[string]*llx.RawData, plugin.Resource, error) {
+	if len(args) > 1 {
+		return args, nil, nil
```
[the remark this answers is not in the document]
Comment (chris-rock) [reply providers/dropbox/resources/dropbox.go]: Fixed in `384d6da`. Changed the fast-path guard from `len(args) > 1` to `len(args) > 2`, matching the CLAUDE.md convention so a `{"__id": ..., "name": ...}` no longer skips the fetch and produces a half-populated resource.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: providers/dropbox/resources/dropbox.go [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+	features, err := client.FeaturesGetValues(&team.FeaturesGetValuesBatchArg{
+		Features: []*team.Feature{{Tagged: dropboxsdk.Tagged{Tag: team.FeatureUploadApiRateLimit}}},
+	})
+	if err == nil {
+		for _, v := range features.Values {
+			if v.UploadApiRateLimit != nil && v.UploadApiRateLimit.Tag == team.UploadApiRateLimitValueLimit {
+				uploadRateLimit = int64(v.UploadApiRateLimit.Limit)
+			}
+		}
+	}
```
[the remark this answers is not in the document]
Comment (chris-rock) [reply providers/dropbox/resources/dropbox.go:64]: Fixed in `384d6da`. `FeaturesGetValues` failures now log at warning level (`log.Warn().Err(err)...`) with a note that `uploadApiRateLimit` falls back to 0 (indistinguishable from unlimited), so operators can diagnose a token missing the required scope.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: providers/dropbox/resources/member.go [hunk 6]
```diff
@@ -0,0 +1,145 @@
+// Copyright Mondoo, Inc. 2024, 2026
```
[the remark this answers is not in the document]
Comment (chris-rock) [reply providers/dropbox/resources/member.go:1]: Fixed in `384d6da`. Added `resources/dropbox_test.go` with table-driven tests covering the pure Go logic. The sharing-policy derivation was extracted from `initDropboxTeam` into a testable `deriveSharingPolicies` helper, now tested across all policy tags plus the nil/empty cases (including `default_team_only`→public and `default_no_one`→not-public). Also added tests for `dbxTimePtr` (nil-preservation and conversion), `firstNonEmpty`, and the generic `pagedFetch` helper (single page, multi-page ordering, and first-page/continuation error propagation). `go test ./...` passes in `providers/dropbox`.

The remaining struct-mapping functions (`newMqlDropboxMember`, device/linked-app mapping) are thin wrappers around `CreateResource` and are covered by interactive verification rather than unit tests.

---
_Generated by [Claude Code](https://claude.ai/code)_
